src/ProtocolZoo/cutoff.jl

```diff
@@ -72,6 +69,8 @@ end
 
             unlock(slot)
         end
-        @yield timeout(prot.sim, prot.period)
+        if !isnothing(prot.period)
+            @yield timeout(prot.sim, prot.period)
+        end
```

Comment:
I do not think this is safe as currently done. Seems like it might lead to infinite repetition of the query. I think there should be some `else` with a wait on tag update.